### PR TITLE
Colors

### PR DESCRIPTION
**Pull request by RunDevelopment**
This adds support for a color data type. Nodes can now produce and consume colors.

image

## TODO

- [x] Add color input element and default color values for color inputs.
- [x] Add image inputs that accept both images and color.
- [x] Remove old "Create Color (Grayscale/RGB/RGBA)" nodes.

### CODE REVIEW

**Comment by joeyballentine** [comment]
Nice. Is the plan to replace the "create color image" nodes with just a single node that accepts a color input?

**Comment by RunDevelopment** [comment]
> Is the plan to replace the "create color image" nodes with just a single node that accepts a color input?

See that "Create Color" node in the first screenshot? 😄 

I just haven't written a migration yet, because I want to add support for color values first. E.g. if the "Create Color Image (RGB)" node doesn't have any incoming connections, we can replace it with a "Create Color" node with an inline color instead of creating an additional "Color From" node.

But good point. I should add that to the TODO list.

**Comment by joeyballentine** [comment]
> See that "Create Color" node in the first screenshot? 😄

Lol I didn't even recognize it

**Review by joeyballentine — COMMENTED** [review]
This looks great. I'm going to test it out first before approving + merging

File: src/renderer/components/inputs/elements/ColorPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    // The react-colorful color picker has a pretty major bug: rounding.
+    // It uses HSV internally to represent color, but it seems to round those values.
+    // This results in the picker sometimes immediately changing the given input color.
+    // This component fixes this behavior by detecting whether the onChange calls are
+    // caused by the user interacting with the color picker or by this bug.
+
+    const pickerRef = useRef<HTMLDivElement>(null);
+    const pickerLastSet = useRef(0);
+
+    // TODO: fix misuse of useMemo
```
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply src/renderer/components/inputs/elements/ColorPicker.tsx:26]: Actually, it seems like they changed the docs. I remember that the docs said that `useMemo` may choose to memoize or not (depending on what React thinks is best) when I first started learning React. But the new docs say: "[React will not throw away the cached value unless there is a specific reason to do that.](https://react.dev/reference/react/useMemo#caveats)" 

So `useMemo` now has strong-enough guarantees for this code to be guaranteed to work properly. That's good to know.

**Review by joeyballentine — APPROVED** [review]

**Pull request merged**